common/logisticspipes/gui/guidebook/GuiGuideBook.java

```diff
@@ -0,0 +1,779 @@
+					if ((chapterIndex + 1) % tile$max == 0) area$currentY += tile$spacing + tile$size;
+					if (chapterIndex == div.getChapters().size() - 1) area$currentY += tile$size;
+				}
+			}
+			return area$currentY;
+		}
+	}
+
+	public class PageState implements State {
+
+		@Override
+		public int draw(Minecraft mc, int mouseX, int mouseY, int yOffset) {
+			String unformattedText = currentPage.page.getText();
+			ArrayList<String> text = StringUtils.splitLines(unformattedText, fontRenderer, area$acrossX);
+			int area$currentY = 0;
+			GlStateManager.pushMatrix();
+			GlStateManager.translate(0, 0, z$text);
+			for (String line : text) {
+				fontRenderer.drawString(line, area$x0, area$y0 + area$currentY + yOffset, 0xFFFFFF);
+				area$currentY += 10;
+			}
+			GlStateManager.popMatrix();
+			GlStateManager.pushMatrix();
+			GlStateManager.translate(0.0F, 0.0F, z$titleButtons);
+			drawPageCount();
+			GlStateManager.popMatrix();
+			return area$currentY;
+		}
+	}
+
+	public class PageInformation {
+
+		@Getter
+		@Setter
+		GuideBookContents.Page page;
+		@Getter
+		@Setter
+		float progress;
+
+		public int getIndex() {
+			return page.getIndex();
+		}
+
+		public int getChapter() {
+			return page.getCindex();
+		}
+
+		public int getDivision() {
+			return page.getDindex();
+		}
+
+		public int getPageCount() {
+			return gbc.getDivision(page.getDindex()).getChapter(page.getCindex()).getNPages();
+		}
+
+		public void changePage(PageInformation pageInfo) {
+			this.page = pageInfo.page;
+			this.progress = pageInfo.progress;
+		}
+
+		public void set(SavedTab tab){
+			page = tab.page.page;
+			progress = tab.page.progress;
+			updateTitle();
+		}
+
+		public void set(MenuItem item) {
+			set(item.getChapter().getDindex(), item.getChapter().getCindex(), 0, 0.0F);
+		}
+
+		public void set(int dindex, int cindex, int index, float progress) {
+			this.page = gbc.getDivision(dindex).getChapter(cindex).getPage(index);
+			this.progress = progress;
+			updateTitle();
+		}
+
+		public void nextPage() {
+			page = gbc.getDivision(page.getDindex()).getChapter(page.getCindex()).getPage(page.getIndex() + 1);
+		}
+
+		public void prevPage() {
+			page = gbc.getDivision(page.getDindex()).getChapter(page.getCindex()).getPage(page.getIndex() - 1);
+		}
+
+		public PageInformation() {
+			page = new GuideBookContents.Page(-1, 0, 0, "");
+			progress = 0.0F;
+		}
+	}
+
+	public class SavedTab {
+
+		public GuiGuideBook.PageInformation page;
+		public GuiGuideBook.State state;
+		public int color;
+
+		public void setTab(GuiGuideBook.PageInformation page, GuiGuideBook.State state) {
+			this.page = page;
+			this.state = state;
+		}
+
+		public SavedTab(PageInformation page, State state) {
+			this.page = page;
+			this.state = state;
+			this.color = 0xFF0000;
+		}
+
+		public SavedTab() {
+			page = new PageInformation();
+			state = new PageState();
+			this.color = 0xFF0000;
+		}
+	}
+
+	public class SavedTabs {
+
+		public ArrayList<SavedTab> tabs;
```

Comment:
Consider making this private and re-create the methods you need. Like `public int size() { return tabs.size() }`